data/model/oci/manifest.py

```diff
@@ -268,44 +317,13 @@ def _create_manifest(
 
     try:
         with db_transaction():
-            # Create the manifest.
-            try:
-                manifest = Manifest.create(
-                    repository=repository_id,
-                    digest=manifest_interface_instance.digest,
-                    media_type=media_type,
-                    manifest_bytes=manifest_interface_instance.bytes.as_encoded_str(),
-                    config_media_type=manifest_interface_instance.config_media_type,
-                    layers_compressed_size=manifest_interface_instance.layers_compressed_size,
-                )
-            except IntegrityError as ie:
-                # NOTE: An IntegrityError means (barring a bug) that the manifest was created by
-                # another caller while we were attempting to create it. Since we need to return
-                # the manifest, we raise a specialized exception here to break out of the
-                # transaction so we can retrieve it.
-                raise _ManifestAlreadyExists(ie)
-
-            # Insert the blobs.
-            blobs_to_insert = [
-                dict(manifest=manifest, repository=repository_id, blob=storage_id)
-                for storage_id in storage_ids
-            ]
-            if blobs_to_insert:
-                try:
-                    ManifestBlob.insert_many(blobs_to_insert).execute()
-                except IntegrityError as ie:
-                    raise _ManifestAlreadyExists(ie)
-
-            # Insert the manifest child rows (if applicable).
+            manifest = create_manifest(repository_id, manifest_interface_instance)
+
+            if storage_ids:
+                connect_blobs(manifest, storage_ids, repository_id)
+
             if child_manifest_rows:
-                children_to_insert = [
-                    dict(manifest=manifest, child_manifest=child_manifest, repository=repository_id)
-                    for child_manifest in list(child_manifest_rows.values())
-                ]
-                try:
-                    ManifestChild.insert_many(children_to_insert).execute()
-                except IntegrityError as ie:
-                    raise _ManifestAlreadyExists(ie)
```

Comment:
this code was extracted into their own functions so that it could be reused elsewhere.